policy-controller/src/admission.rs

```diff
@@ -346,6 +347,14 @@ impl Validate<ServerSpec> for Admission {
             }
         }
 
+        if let Some(Err(_)) = spec
+            .clone()
+            .access_policy
+            .map(|p| p.parse::<index::DefaultPolicy>())
+        {
+            bail!("Invalid accessPolicy \"{}\"", spec.access_policy.unwrap());
+        }
+
```

Comment:
suggestion: I would perhaps re-write this slightly to make it a bit simpler and to avoid cloning the spec. In my mind, whatever spec we're dealing with, has already been moved into the function (so its owned by `validate`), so we could definitely avoid it either by using a reference to the inner field or by re-writing this slightly.

```rust
        if let Some(policy) = spec.access_policy {
            policy.parse::<index::DefaultPolicy>().map_err(|err| {
                anyhow::anyhow!(
                    "Server spec '{ns}/{name}' has invalid 'access_policy' field: {err}"
                )
            })?;
        }
```

This might preserve the current intention by avoiding a clone and also using the error message generated from the call to `parse`. We can either return the error as-is or decorate it with some additional information (by using `anyhow!`).

Reply:
Good call on not relying on clone() :+1: 
I'll use your snippet without the ns/name part, to keep the error succinct, given it should be know what resource the error is about.